lib/fog/compute/kubevirt.rb

```diff
@@ -335,51 +338,69 @@ def populate_notice_attributes(object, notice)
         end
 
         #
-        # Lazily creates the a client for the given Kubernetes API path and version.
+        # Lazily creates the a client for the given Kubernetes API path.
         #
         # @param path [String] The Kubernetes API path.
-        # @param version [String] The Kubernetes API version.
         # @return [Kubeclient::Client] The client for the given path and version.
         #
-        def create_client(path, version)
-          # Return the client immediately if it has been created before:
-          key = path + '/' + version
-          client = @clients[key]
-          return client if client
-
+        def create_client(path)
           # Create the client and save it:
           url = URI::Generic.build(
             :scheme => 'https',
             :host   => @host,
             :port   => @port,
             :path   => path
           )
+
+          version = detectVersion(url.to_s, @opts[:ssl_options])
+          key = path + '/' + version
+          client = @clients[key]
+          return client if client
+
           client = Kubeclient::Client.new(
             url.to_s,
             version,
             @opts
           )
-          wrapped_client = ExceptionWrapper.new(client)
+          wrapped_client = ExceptionWrapper.new(client, version)
           @clients[key] = wrapped_client
 
           # Return the client:
           wrapped_client
         end
 
+        def detectVersion(url, ssl_options)
+          options = {
+            ssl_ca_file: ssl_options[:ca_file],
+            ssl_cert_store: ssl_options[:cert_store],
+            verify_ssl: ssl_options[:verify_ssl],
+            ssl_client_cert: ssl_options[:client_cert],
+            ssl_client_key: ssl_options[:client_key],
+          }
+
+          response = JSON.parse(RestClient::Resource.new(url, options).get)
```

Comment:
This fails with:
```
undefined method `parse' for Fog::JSON:Module
```
Please add `::` before: `::JSON.parse...`

Reply:
I do not get this issue when testing fog-kubevirt but will update.

Reply:
Same here, `bundle exec rake` finished without issues. It was detected on foreman-kubevirt plugin.